### DISCUSSION TITLE
The best practice of default export & file naming?

### DISCUSSION THREAD

**Discussion by EnixCoda** [reduxjs/redux-toolkit#2772] [Polls]
I noticed that the example code in the doc is exporting **reducer** from a file whose name ends with **slice**. Then the import statements for it would be like `import reducer from './slice'`. This does not look very intuitive to me. So I'd like to confirm if this was intended and see your opinions about this.

## Examples from doc

<img width="946" alt="image" src="https://user-images.githubusercontent.com/7480839/195825868-41c9ad07-0c56-4b65-b691-2ef8f578ef91.png">

https://redux-toolkit.js.org/usage/usage-guide#using-createentityadapter-with-normalization-libraries

And its usage might be `import usersReducer from 'features/users/usersSlice'` 🤨

Note: there could be more exported members in the file, like actions and selectors. I do not list them in the options below but they shall exist in the file as well.

Thank you!

**Comment by phryneas** [comment]
There is not really a a question "which we prefer". 
It has been the standard for the last 7 years, there are myriads of tutorials, videos and blog posts using that style. If we now derived from it, it would be horribly confusing for newcomers that jump between multiple sources (many people at that stage do not know the details of imports & exports yet).

Personally I think default exports should be removed from the spec, but for the same reason that is not going to happen we will also not change this best practice.

Of course you can do in your own code base as you prefer it.

**Reply by markerikson** [reply]
To add to this: this is effectively an approach known as the "ducks pattern" that was first described in mid-2015, within a month of Redux hitting 1.0:

https://github.com/erikras/ducks-modular-redux

That approach was already decently known in the Redux community by the time we developed Redux Toolkit, and it's a fairly reasonable pattern.  It also lends itself to being able to do something as `import * as someSliceActions from "./someSlice"` (if you actually wanted to do that).

As Lenz said: it's your code, you can do whatever you want to.  But the idea of default-exporting a reducer has been in the docs for years, so it's not like this is a new thing.
